assets/blocks/frontend.js

```diff
@@ -46,10 +46,17 @@ domReady( () => {
 			const collapsed = content.classList.toggle( 'collapsed' );
 
 			if ( ! collapsed ) {
+				content.style.visibility = 'visible';
 				content.style.maxHeight = originalHeight;
 			} else {
 				content.style.maxHeight = '0px';
 			}
 		} );
+
+		content.addEventListener( 'transitionend', () => {
+			if ( content.classList.contains( 'collapsed' ) ) {
+				content.style.visibility = 'hidden';
+			}
+		} );
```

Comment:
I try hard to not overthink things and this is what comes out of it. :D 

The `transitionend` event is definitely a better approach than trying to time it.

Surprised to see you using the `visibility: hidden;` CSS though. Didn't realize it could also work to avoid focus capturing of the elements.